plugin-server/tests/worker/ingestion/cookielessServerHashStep.test.ts

```diff
@@ -0,0 +1,371 @@
+                },
+                site_url: 'https://example.com',
+                team_id: teamId,
+                now: now.toISOString(),
+                uuid: new UUID7(now.getTime()).toString(),
+                ip: null,
+            })
+            identifyEventABitLater = deepFreeze({
+                ...identifyEvent,
+                now: aBitLater.toISOString(),
+                uuid: new UUID7(aBitLater.getTime()).toString(),
+            })
+            postIdentifyEvent = deepFreeze({
+                event: 'test event',
+                distinct_id: identifiedDistinctId,
+                properties: {
+                    [COOKIELESS_MODE_FLAG_PROPERTY]: true,
+                    $host: 'https://example.com',
+                    $raw_user_agent: userAgent,
+                    $ip: '1.2.3.4',
+                },
+                site_url: 'https://example.com',
+                team_id: teamId,
+                now: now.toISOString(),
+                uuid: new UUID7(now.getTime()).toString(),
+                ip: null,
+            })
+            aliasEvent = deepFreeze({
+                ...event,
+                event: '$create_alias',
+            })
+            mergeDangerouslyEvent = deepFreeze({
+                ...event,
+                event: '$merge_dangerously',
+            })
+            nonCookielessEvent = deepFreeze({
+                ...event,
+                properties: {
+                    $host: 'https://example.com',
+                    $raw_user_agent: userAgent,
+                },
+            })
+        })
+
+        // tests that are shared between both modes
+        describe.each([
+            ['stateless', CookielessServerHashMode.Stateless],
+            ['stateful', CookielessServerHashMode.Stateful],
+        ])('common (%s)', (_, mode) => {
+            beforeEach(async () => {
+                await setModeForTeam(mode, teamId)
+            })
+            it('should give an event a distinct id and session id ', async () => {
+                const [actual] = await cookielessServerHashStep(hub, event)
+
+                if (!actual?.properties) {
+                    throw new Error('no event or properties')
+                }
+                expect(actual.distinct_id).not.toEqual(COOKIELESS_SENTINEL_VALUE)
+                expect(actual.distinct_id.startsWith('cookieless_')).toBe(true)
+                expect(actual.properties.$session_id).toBeDefined()
+            })
+            it('should give the same session id and distinct id to events with the same hash properties and within the same day and session timeout period', async () => {
+                const [actual1] = await cookielessServerHashStep(hub, event)
+                const [actual2] = await cookielessServerHashStep(hub, eventABitLater)
+
+                if (!actual1?.properties || !actual2?.properties) {
+                    throw new Error('no event or properties')
+                }
```

Comment:
Same as above, I'd refrain from throwing errors and would use asserts instead

```suggestion
                expect(actual1?.properties).not.toBeNull()
                expect(actual2?.properties).not.toBeNull()
```

Same for the ones below